src/SDK/Resource/ResourceInfo.php

```diff
@@ -62,6 +62,7 @@ public function serialize(): string
      * resource, the value of the updating resource MUST be picked (even if the updated value is empty)
      *
      * @see https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/resource/sdk.md#merge
+     * @todo can we optimize this to avoid re-validating the attributes on merge?
```

Comment:
Created an issue for this: https://github.com/open-telemetry/opentelemetry-php/issues/1212